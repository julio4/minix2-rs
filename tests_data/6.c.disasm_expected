0000: 31ed          xor bp, bp
0002: 89e3          mov bx, sp
0004: 8b07          mov ax, [bx]
0006: 8d5702        lea dx, [bx+2]
0009: 8d4f04        lea cx, [bx+4]
000c: 01c1          add cx, ax
000e: 01c1          add cx, ax
0010: bb0800        mov bx, 0008
0013: 81fb0c00      cmp bx, 000c
0017: 730f          jnb 0028
0019: f6c301        test bl, 1
001c: 750a          jne 0028
001e: 813f5353      cmp [bx], 5353
0022: 7504          jne 0028
0024: 891e0200      mov [0002], bx
0028: 8b1e0200      mov bx, [0002]
002c: 890f          mov [bx], cx
002e: 51            push cx
002f: 52            push dx
0030: 50            push ax
0031: e80500        call 0039
0034: 50            push ax
0035: e88300        call 00bb
0038: f4            hlt
0039: 55            push bp
003a: 89e5          mov bp, sp
003c: b8e514        mov ax, 14e5
003f: 50            push ax
0040: e80400        call 0047
0043: 5b            pop bx
0044: e93e01        jmp 0185
0047: 55            push bp
0048: 89e5          mov bp, sp
004a: 56            push si
004b: 57            push di
004c: 8b7e04        mov di, [bp+4]
004f: 09ff          or di, di
0051: 7d05          jnl 0058
0053: b83100        mov ax, 0031
0056: eb03          jmp short 005b
0058: b83000        mov ax, 0030
005b: 50            push ax
005c: e82600        call 0085
005f: 5b            pop bx
0060: 31f6          xor si, si
0062: 83fe05        cmp si, 5
0065: 7d1b          jnl 0082
0067: b90c00        mov cx, 000c
006a: 89fa          mov dx, di
006c: d3fa          sar dx, cl
006e: 81e20700      and dx, 0007
0072: 83c230        add dx, 30
0075: 52            push dx
0076: e80c00        call 0085
0079: 5b            pop bx
007a: b90300        mov cx, 0003
007d: d3e7          shl di, cl
007f: 46            inc si
0080: ebe0          jmp short 0062
0082: e9fe00        jmp 0183
0085: 55            push bp
0086: 89e5          mov bp, sp
0088: b80100        mov ax, 0001
008b: 50            push ax
008c: 8d4604        lea ax, [bp+4]
008f: 50            push ax
0090: b80100        mov ax, 0001
0093: 50            push ax
0094: e84500        call 00dc
0097: 83c406        add sp, 6
009a: 8a4604        mov al, [bp+4]
009d: 98            cbw
009e: e9e400        jmp 0185
00a1: 55            push bp
00a2: 89e5          mov bp, sp
00a4: 56            push si
00a5: 8b360400      mov si, [0004]
00a9: 4e            dec si
00aa: 7c0c          jl 00b8
00ac: 89f3          mov bx, si
00ae: d1e3          shl bx, 1
00b0: 8b9f0e00      mov bx, [bx+e]
00b4: ffd3          call bx
00b6: ebf1          jmp short 00a9
00b8: e9c900        jmp 0184
00bb: 55            push bp
00bc: 89e5          mov bp, sp
00be: e8e0ff        call 00a1
00c1: 833e060000    cmp [0006], 0
00c6: 7406          je 00ce
00c8: 8b1e0600      mov bx, [0006]
00cc: ffd3          call bx
00ce: ff7604        push [bp+4]
00d1: e80400        call 00d8
00d4: 5b            pop bx
00d5: e9ad00        jmp 0185
00d8: e90500        jmp 00e0
00db: 00e9          add cl, ch
00dd: 2100          and [bx+si], ax
00df: 005589        add [di-77], dl
00e2: e583          in ax, 83
00e4: ec            in al, dx
00e5: 188b5604      sbb [bp+di+456], cl
00e9: 8956ec        mov [bp-14], dx
00ec: 8d46e8        lea ax, [bp-18]
00ef: 50            push ax
00f0: b80100        mov ax, 0001
00f3: 50            push ax
00f4: 31c0          xor ax, ax
00f6: 50            push ax
00f7: e83300        call 012d
00fa: 83c406        add sp, 6
00fd: e98500        jmp 0185
0100: 55            push bp
0101: 89e5          mov bp, sp
0103: 83ec18        sub sp, 18
0106: 8b5604        mov dx, [bp+4]
0109: 8956ec        mov [bp-14], dx
010c: 8b5608        mov dx, [bp+8]
010f: 8956ee        mov [bp-12], dx
0112: 8b5606        mov dx, [bp+6]
0115: 8956f2        mov [bp-e], dx
0118: 8d46e8        lea ax, [bp-18]
011b: 50            push ax
011c: b80400        mov ax, 0004
011f: 50            push ax
0120: b80100        mov ax, 0001
0123: 50            push ax
0124: e80600        call 012d
0127: 83c406        add sp, 6
012a: e95800        jmp 0185
012d: 55            push bp
012e: 89e5          mov bp, sp
0130: 56            push si
0131: 57            push di
0132: 8b7608        mov si, [bp+8]
0135: 8b5606        mov dx, [bp+6]
0138: 895402        mov [si+2], dx
013b: 56            push si
013c: ff7604        push [bp+4]
013f: e82f00        call 0171
0142: 5b            pop bx
0143: 5b            pop bx
0144: 89c7          mov di, ax
0146: 09ff          or di, di
0148: 7403          je 014d
014a: 897c02        mov [si+2], di
014d: 837c0200      cmp [si+2], 0
0151: 7d0e          jnl 0161
0153: 8b5402        mov dx, [si+2]
0156: f7da          neg dx
0158: 89160a00      mov [000a], dx
015c: b8ffff        mov ax, ffff
015f: eb03          jmp short 0164
0161: 8b4402        mov ax, [si+2]
0164: e91c00        jmp 0183
0167: b90100        mov cx, 0001
016a: eb0a          jmp short 0176
016c: b90200        mov cx, 0002
016f: eb05          jmp short 0176
0171: b90300        mov cx, 0003
0174: eb00          jmp short 0176
0176: 55            push bp
0177: 89e5          mov bp, sp
0179: 8b4604        mov ax, [bp+4]
017c: 8b5e06        mov bx, [bp+6]
017f: cd20          int 20
0181: 5d            pop bp
0182: c3            ret
0183: 5f            pop di
0184: 5e            pop si
0185: 89ec          mov sp, bp
0187: 5d            pop bp
0188: c3            ret
0189: 5e            pop si
018a: 5f            pop di
018b: ebf8          jmp short 0185
018d: 0000          add [bx+si], al
018f: 00            (undefined)
