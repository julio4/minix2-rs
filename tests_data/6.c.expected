0000: 55            push bp
0001: 89e5          mov bp, sp
0003: b8e514        mov ax, 14e5
0006: 50            push ax
0007: e80400        call 000e
000a: 5b            pop bx
000b: e9e300        jmp 00f1
000e: 55            push bp
000f: 89e5          mov bp, sp
0011: 56            push si
0012: 57            push di
0013: 8b7e04        mov di, [bp+4]
0016: 09ff          or di, di
0018: 7d05          jnl 001f
001a: b83100        mov ax, 0031
001d: eb03          jmp short 0022
001f: b83000        mov ax, 0030
0022: 50            push ax
0023: e82600        call 004c
0026: 5b            pop bx
0027: 31f6          xor si, si
0029: 83fe05        cmp si, 5
002c: 7d1b          jnl 0049
002e: b90c00        mov cx, 000c
0031: 89fa          mov dx, di
0033: d3fa          sar dx, cl
0035: 81e20700      and dx, 0007
0039: 83c230        add dx, 30
003c: 52            push dx
003d: e80c00        call 004c
0040: 5b            pop bx
0041: b90300        mov cx, 0003
0044: d3e7          shl di, cl
0046: 46            inc si
0047: ebe0          jmp short 0029
0049: e9a300        jmp 00ef
004c: 55            push bp
004d: 89e5          mov bp, sp
004f: b80100        mov ax, 0001
0052: 50            push ax
0053: 8d4604        lea ax, [bp+4]
0056: 50            push ax
0057: b80100        mov ax, 0001
005a: 50            push ax
005b: e80a00        call 0068
005e: 83c406        add sp, 6
0061: 8a4604        mov al, [bp+4]
0064: 98            cbw
0065: e98900        jmp 00f1
0068: e90100        jmp 006c
006b: 005589        add [di-77], dl
006e: e583          in ax, 83
0070: ec            in al, dx
0071: 188b5604      sbb [bp+di+456], cl
0075: 8956ec        mov [bp-14], dx
0078: 8b5608        mov dx, [bp+8]
007b: 8956ee        mov [bp-12], dx
007e: 8b5606        mov dx, [bp+6]
0081: 8956f2        mov [bp-e], dx
0084: 8d46e8        lea ax, [bp-18]
0087: 50            push ax
0088: b80400        mov ax, 0004
008b: 50            push ax
008c: b80100        mov ax, 0001
008f: 50            push ax
0090: e80600        call 0099
0093: 83c406        add sp, 6
0096: e95800        jmp 00f1
0099: 55            push bp
009a: 89e5          mov bp, sp
009c: 56            push si
009d: 57            push di
009e: 8b7608        mov si, [bp+8]
00a1: 8b5606        mov dx, [bp+6]
00a4: 895402        mov [si+2], dx
00a7: 56            push si
00a8: ff7604        push [bp+4]
00ab: e82f00        call 00dd
00ae: 5b            pop bx
00af: 5b            pop bx
00b0: 89c7          mov di, ax
00b2: 09ff          or di, di
00b4: 7403          je 00b9
00b6: 897c02        mov [si+2], di
00b9: 837c0200      cmp [si+2], 0
00bd: 7d0e          jnl 00cd
00bf: 8b5402        mov dx, [si+2]
00c2: f7da          neg dx
00c4: 89160000      mov [0000], dx
00c8: b8ffff        mov ax, ffff
00cb: eb03          jmp short 00d0
00cd: 8b4402        mov ax, [si+2]
00d0: e91c00        jmp 00ef
00d3: b90100        mov cx, 0001
00d6: eb0a          jmp short 00e2
00d8: b90200        mov cx, 0002
00db: eb05          jmp short 00e2
00dd: b90300        mov cx, 0003
00e0: eb00          jmp short 00e2
00e2: 55            push bp
00e3: 89e5          mov bp, sp
00e5: 8b4604        mov ax, [bp+4]
00e8: 8b5e06        mov bx, [bp+6]
00eb: cd20          int 20
00ed: 5d            pop bp
00ee: c3            ret
00ef: 5f            pop di
00f0: 5e            pop si
00f1: 89ec          mov sp, bp
00f3: 5d            pop bp
00f4: c3            ret
00f5: 5e            pop si
00f6: 5f            pop di
00f7: ebf8          jmp short 00f1
00f9: 0000          add [bx+si], al
00fb: 0000          add [bx+si], al
00fd: 0000          add [bx+si], al
00ff: 00            (undefined)
